### DISCUSSION TITLE
Swift function signature for `addOnConfigUpdateListener`

### DISCUSSION THREAD

**Discussion by teejayhh** [firebase/firebase-ios-sdk#11513] [Q&A] [answered]
Hello, we wrote a swift protocol to mock the RemoteConfig during our tests. I wanted to implement the new live update feature but I struggle to create the swift function signature to conform to `RemoteConfig`

The following is what I could come up with but it doesn't conform
```
protocol RemotelyConfigurable {
   func addOnConfigUpdateListener(listener: (RemoteConfigUpdate, Error?) -> Void) -> ConfigUpdateListenerRegistration
}

extension RemoteConfig: RemotelyConfigurable {}
```

I have never learned objective-c so and Bard as well as Bing(GPT) were not helpful. 

I hope someone can help me with this.

**Comment by paulb777** [comment]
You might need to add `@objc`. See https://stackoverflow.com/questions/24078242/importing-a-swift-protocol-in-objective-c-class.

Also for RemoteConfig testing, you may want to check out the fake console test infrastructure at https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseRemoteConfigSwift/Tests

**Reply by teejayhh** [reply]
Thank you for your answer, I learned something, unfortunately this does not fix the conformance issue. As for the testing aspect we use the protocol for code coverage, to check if everything has been called when launching the app. we do not test responses or anything like that as that is not our domain.

**Comment by silvgabriel** [comment]
@teejayhh 
Did you fix this?
I'm going through the same problem

**Reply by teejayhh** [reply, marked as answer]
yep this is it.

```
func addOnConfigUpdateListener(remoteConfigUpdateCompletion listener: @escaping (RemoteConfigUpdate?, Error?) -> Void) -> ConfigUpdateListenerRegistration
```

Sorry I wanted to post it here but forgot. 

Hope this helps

**Reply by silvgabriel** [reply]
It worked!

I was trying:
```
func addOnConfigUpdateListener(remoteConfigUpdateCompletion: @escaping (RemoteConfigUpdate?, Error?) -> Void) -> ConfigUpdateListenerRegistration
```

Thank you @teejayhh
